### PR TITLE
Make sure we use newly built runtime to generate lock file for tests

### PR DESCRIPTION
**Pull request by ChengTian**
@davidfowl 

Tested locally with experimental lock file format change (bumping version up by 1)

### CODE REVIEW

**Comment by ChengTian** [comment]
Each sample-app-based test should have a copy of the sample app because, when parallel testing is turned on, we don't want multiple `dnu restore` writing to one lock file.

**Comment by davidfowl** [comment]
When the ci machine is happy

**Comment by ChengTian** [comment]
@davidfowl , Happy CI X 3

File: test/Bootstrapper.FunctionalTests/BootstrapperTests.cs [hunk 1]
```diff
@@ -151,19 +157,25 @@ public void BootstrapperInvokesAssemblyWithInferredAppBaseAndLibPathOnClr(string
...
                 var exitCode = DnuTestUtils.ExecDnu(
                     runtimeHomeDir,
                     subcommand: "build",
-                    arguments: string.Format("{0} --configuration=Release --out {1}", sampleAppRoot, tempDir.DirPath));
+                    arguments: string.Format("{0} --configuration=Release --out {1}", sampleAppRoot, tempDir),
+                    stdOut: out stdOut,
+                    stdErr: out stdErr);
 
-                Assert.Equal(0, exitCode);
+                if (exitCode != 0)
```
Comment (victorhurdugaci) [inline test/Bootstrapper.FunctionalTests/BootstrapperTests.cs]: Do you want to continue running if this failed?

Comment (ChengTian) [inline test/Bootstrapper.FunctionalTests/BootstrapperTests.cs]: Yes, the failure will be caught by later assertion. We are not testing `dnu restore` here so we don't assert on the exit code of `dnu restore`

**Pull request merged**